File: Day 1/Solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func IncreasingDepthsPart1(depths *[]int) int {

	output := 0

	for previousIndex, depth := range (*depths)[1:] {

		if depth > (*depths)[previousIndex] {
			output += 1
		}

	}

	return output
}

func IncreasingDepthsPart2(depths *[]int) int {

	output := 0

	for previousIndex, depth := range (*depths)[3:] {

		if depth > (*depths)[previousIndex] {
			output += 1
		}

	}

	return output
}

func main() {

	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error reading in input file", err)
	}

	defer file.Close()

	scanner := bufio.NewScanner(file)

	var depths []int

	for scanner.Scan() {
		num, _ := strconv.Atoi(scanner.Text())
		depths = append(depths, num)
	}

	fmt.Println(IncreasingDepthsPart1(&depths))
	fmt.Println(IncreasingDepthsPart2(&depths))
}

File: Day 1/Solution.py

def IncreasingDepthsPart1():

    file = open("input.txt")

    depths = []

    # the values are read in with newline characters, remove them by casting them to ints
    for value in file.readlines():
        depths.append(int(value))

    prev = depths[0]

    output = 0

    for i in range(1, len(depths)):

        if depths[i] > prev:
            output += 1
        
        prev = depths[i]

    return output

def IncreasingDepthsPart2():

    file = open("input.txt")

    depths = []

    # the values are read in with newline characters, remove them by casting them to ints
    for value in file.readlines():
        depths.append(int(value))

    prev = sum(depths[:2])
    output = 0

    for i in range(1, len(depths) - 2):

        if sum(depths[i:i+3]) > prev:
            output += 1
        
        prev = sum(depths[i:i+3])


    return output

print(IncreasingDepthsPart1())
print(IncreasingDepthsPart2())